core/server/master/src/main/java/alluxio/master/file/meta/InodeTreePersistentState.java

```diff
@@ -637,6 +642,7 @@ private void applyCreateInode(MutableInode<?> inode) {
       mToBePersistedIds.clear();
 
       updateToBePersistedIds(inode);
+      mRoot = null;
```

Comment:
The mRoot should be set here. This method looks like the only place where the root can be created in the state machine. If you set the value in the writer, the reader's burden on race condition will be much alleviated.

Reply:
Done, set the mRoot here